tests/context.rs

```diff
@@ -0,0 +1,192 @@
+#![cfg(test)]
+
+use capataz::context::*;
+use tokio::time::{self, Duration};
+use tokio_test::{assert_pending, assert_ready, task};
+
+#[tokio::test]
+async fn test_with_timeout_completion() {
+    time::pause();
+
+    let ctx0 = Context::new();
+
+    let (ctx, _cancel_timeout) = Context::with_timeout(&ctx0, Duration::from_millis(100));
+
+    let mut fut = task::spawn(ctx.done);
+    assert_pending!(fut.poll());
+
+    // fast-forward
+    time::advance(Duration::from_millis(101)).await;
+    assert_ready!(fut.poll());
+}
+
+#[tokio::test]
+async fn test_with_timeout_nested_child_completion() {
+    use tokio::time;
+    use tokio_test::{assert_pending, assert_ready, task};
```

Comment:
Redundant imports. Also applies to all the other imports down in the other functions.